homeassistant/components/influxdb/config_flow.py

```diff
@@ -0,0 +1,275 @@
+            errors = {"base": "ssl_error"}
+        elif "database not found" in ex.args[0]:
+            errors = {"base": "invalid_database"}
+        elif "authorization failed" in ex.args[0]:
+            errors = {"base": "invalid_auth"}
+        elif "token" in ex.args[0]:
+            errors = {"base": "invalid_config"}
+        else:
+            errors = {"base": "cannot_connect"}
+    except Exception:
+        _LOGGER.exception("Unknown error")
+        errors = {"base": "unknown"}
+
+    return errors
+
+
+async def _save_uploaded_cert_file(hass: HomeAssistant, uploaded_file_id: str) -> Path:
+    """Move the uploaded file to storage directory."""
+
+    def _process_upload() -> Path:
+        with process_uploaded_file(hass, uploaded_file_id) as file_path:
+            dest_path = Path(hass.config.path(STORAGE_DIR, DOMAIN))
+            dest_path.mkdir(exist_ok=True)
+            file_name = f"influxdb{file_path.suffix}"
+            dest_file = dest_path / file_name
+            shutil.move(file_path, dest_file)
+        return dest_file
+
+    return await hass.async_add_executor_job(_process_upload)
+
+
+class InfluxDBConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for InfluxDB."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when user initializes a integration."""
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["configure_v1", "configure_v2"],
+        )
+
+    async def async_step_configure_v1(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when user configures InfluxDB v1."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            url = URL(user_input[CONF_URL])
+            data = {
+                CONF_API_VERSION: DEFAULT_API_VERSION,
+                CONF_HOST: url.host,
+                CONF_PORT: url.port,
+                CONF_USERNAME: user_input.get(CONF_USERNAME),
+                CONF_PASSWORD: user_input.get(CONF_PASSWORD),
+                CONF_DB_NAME: user_input[CONF_DB_NAME],
+                CONF_SSL: url.scheme == "https",
+                CONF_PATH: url.path,
+                CONF_VERIFY_SSL: user_input[CONF_VERIFY_SSL],
+            }
+            if (cert := user_input.get(CONF_SSL_CA_CERT)) is not None:
+                path = await _save_uploaded_cert_file(self.hass, cert)
+                data[CONF_SSL_CA_CERT] = str(path)
+            errors = await _validate_influxdb_connection(self.hass, data)
+
+            if not errors:
+                title = f"{data[CONF_DB_NAME]} ({data[CONF_HOST]})"
+                return self.async_create_entry(title=title, data=data)
+
+        schema = INFLUXDB_V1_SCHEMA
+
+        return self.async_show_form(
+            step_id="configure_v1", data_schema=schema, errors=errors
+        )
```

Comment:
Use `add_suggested_values_to_schema` so the user doesn't need to fill in the entire form again